wings-software/wingsui-eslint-rules
# feat: [PL-3359]: Added two new lint rules, to disable usage of componentWillMount and await before this.setState

To test it out locally, 

1. clone the new repo, hit yarn link
2. Go to wingsui, hit yarn link `eslint-plugin-wingsui-rules`
3. open .eslintrc and update the following lines : 



```
"plugins": ["react", "@typescript-eslint/eslint-plugin", "wingsui-rules"], //wingsui-rules is the new entry


"rules": {
    // Common js rules
    "wingsui-rules/no-await-before-setstate": 2,
    "wingsui-rules/no-component-will-mount": 2,
}

```

4. Try using componentWillMount or await this.setState and see an error message like below : 

<img width="1252" alt="Screenshot 2019-09-15 at 1 09 24 PM" src="https://user-images.githubusercontent.com/51908439/64918319-24097800-d7ba-11e9-9d5b-312abfb534d3.png">


References : 
https://blog.webiny.com/create-custom-eslint-rules-in-2-minutes-e3d41cb6a9a0
https://astexplorer.net/
https://github.com/eslint/generator-eslint
